### PR TITLE
[E2E] Initial commit of e2e test utils

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

This PR introduces KubeClient, and AppManager, which are the helpers to deploy app and create service, and clean up deployment and services by using k8s client-go. 

K8S fake client and reactor are used for the unit-tests for these implementations.

When it comes to helloworld_test.go in this pr. The current implementation is to show how to use appmanager. The next PR will include test framework to simplify the test setup and cleanup steps. then I will rewrite helloworld_test.go with new test framework.

## Issue reference

#734 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: tests/platforms/kubernetes/appmanager.go [hunk 1]
```diff
@@ -12,101 +12,93 @@ import (
...
+	"github.com/dapr/dapr/tests/utils"
 	appsv1 "k8s.io/api/apps/v1"
 	apiv1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 const (
+	// MiniKubeIPEnvVar is the environment variable name which will have Minikube node IP
+	MiniKubeIPEnvVar = "MINIKUBE_IP"
```
Comment (mmisztal1980) [inline tests/platforms/kubernetes/appmanager.go]: do the utils assume Minikube usage? What docker for desktop k8s is used?

Comment (youngbupark) [reply tests/platforms/kubernetes/appmanager.go]: This is the initial commit of e2e test utils. We're targeting to only minikube for local development experience in this iteration. kind, k8s in docker desktop, etc will be next.

Comment (youngbupark) [reply tests/platforms/kubernetes/appmanager.go]: @mmisztal1980 If you are interested in the spec and the progress of e2e test, please see the below links
* https://github.com/dapr/dapr/issues/335#issuecomment-548217582
* https://github.com/orgs/dapr/projects/9#card-28802684

Comment (mmisztal1980) [reply tests/platforms/kubernetes/appmanager.go]: I am, cheers!

File: tests/README.md [hunk 2]
```diff
@@ -1,9 +1,3 @@
...
 
-> This is still working in progress.
-
-This contains the following tests and testing docs for Dapr:
-
-- [Test Apps](./apps)
-- [End to end tests](./e2e)
-- [Platform-dependent test utils](./platforms)
\ No newline at end of file
+> This is working in progress.
```
Comment (yaron2) [inline tests/README.md]: Needs to be `work in progress`

File: tests/e2e/helloworld_test.go [hunk 3]
```diff
@@ -16,10 +16,11 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// NOTE!! WORKING IN PROGRESS.
+// !!NOTE!! WORKING IN PROGRESS. !!NOTE!!
```
Comment (yaron2) [inline tests/e2e/helloworld_test.go]: `WORK IN PROGRESS`

File: tests/platforms/kubernetes/appmanager.go [hunk 4]
```diff
@@ -108,7 +108,7 @@ func (m *AppManager) ValidiateSideCar(app utils.AppDescription) (bool, error) {
 	}
 
 	if len(podList.Items) != int(app.Replicas) {
-		return false, fmt.Errorf("Number of Pods for %s must be %d Pods but %d Pods", app.AppName, app.Replicas, len(podList.Items))
+		return false, fmt.Errorf("number of Pods for %s must be %d Pods but %d Pods", app.AppName, app.Replicas, len(podList.Items))
```
Comment (yaron2) [inline tests/platforms/kubernetes/appmanager.go]: Rephrase to `expected number of pods for %s: %d, received %d`

File: tests/platforms/kubernetes/appmanager.go [hunk 5]
```diff
@@ -120,7 +120,7 @@ func (m *AppManager) ValidiateSideCar(app utils.AppDescription) (bool, error) {
 			}
 		}
 		if !daprdFound {
-			return false, fmt.Errorf("Cannot find dapr sidecar in %s pod", pod.Name)
+			return false, fmt.Errorf("cannot find dapr sidecar in %s pod", pod.Name)
```
Comment (yaron2) [inline tests/platforms/kubernetes/appmanager.go]: Rephrase to `cannot find dapr sidecar in pod %s`

File: tests/platforms/kubernetes/appmanager.go [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+	})
+	if err != nil {
+		return false, err
+	}
+
+	if len(podList.Items) != int(app.Replicas) {
+		return false, fmt.Errorf("number of Pods for %s must be %d Pods but %d Pods", app.AppName, app.Replicas, len(podList.Items))
+	}
+
+	// All testapp pods must have daprd sidecar
```
Comment (yaron2) [inline tests/platforms/kubernetes/appmanager.go]: `test app`

File: tests/platforms/kubernetes/kubeobj.go [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// BuildServiceObject creates the Kubernetes Service Object for dapr test app
+func BuildServiceObject(namespace string, appDesc utils.AppDescription) *apiv1.Service {
```
Comment (yaron2) [inline tests/platforms/kubernetes/kubeobj.go]: Does this method need to be exported?

File: tests/utils/app_description.go [hunk 8]
```diff
@@ -0,0 +1,11 @@
+package utils
+
+// AppDescription holds the deployment information of test app
+type AppDescription struct {
```
Comment (yaron2) [inline tests/utils/app_description.go]: Don't we need the app port here too?

File: tests/README.md (hunk 2 above)
Comment (youngbupark) [reply tests/README.md]: corrected.

File: tests/e2e/helloworld_test.go (hunk 3 above)
Comment (youngbupark) [reply tests/e2e/helloworld_test.go]: corrected.

File: tests/platforms/kubernetes/appmanager.go (hunk 4 above)
Comment (youngbupark) [reply tests/platforms/kubernetes/appmanager.go]: Fixed.

File: tests/platforms/kubernetes/appmanager.go (hunk 6 above)
Comment (youngbupark) [reply tests/platforms/kubernetes/appmanager.go]: Fixed.

File: tests/platforms/kubernetes/appmanager.go (hunk 5 above)
Comment (youngbupark) [reply tests/platforms/kubernetes/appmanager.go]: Fixed.

File: tests/platforms/kubernetes/kubeobj.go (hunk 7 above)
Comment (youngbupark) [reply tests/platforms/kubernetes/kubeobj.go]: No. Made it private.

File: tests/utils/app_description.go (hunk 8 above)
Comment (youngbupark) [reply tests/utils/app_description.go]: For now, I will use the default app port 3000 and http protocol.

I created [this issue](https://github.com/orgs/dapr/projects/9#card-28915569) to support the custom app-port and protocol.

Comment (yaron2) [reply tests/utils/app_description.go]: ok

**Comment by yaron2** [comment]
Ready to merge?

**Pull request merged**